educe/rst_dt/rst_wsj_corpus.py

```diff
@@ -0,0 +1,158 @@
+    """
+    with codecs.open(f, 'r', 'utf-8') as f:
+        clean_txt, offset = _load_rst_wsj_corpus_edus_file(f)
+    return (clean_txt, offset)
+
+
+# TODO what the _text_ functions in this module should return is
+# clean text plus the extractible annotation,
+# i.e. paragraph spans (if any) and sentence spans,
+# *on the clean text*
+
+##############
+# file## files
+##############
+
+# TODO: maybe don't generate Paragraph, as there is no paragraph marking
+FIL_SEP_PARA = '\n\n'  # probably useless
+FIL_SEP_SENT = '\n  '
+
+# TODO move to more relevant place (?)
+# TODO use this info :-)
+file_to_ptb = {'file1': 'wsj_0764',
+}
+
+
+def _load_rst_wsj_corpus_text_file_file(f):
+    """Actually do load"""
+    text = f.read()
+
+    start = 0
+    sent_id = 0
+    output_paras = []
+    for para_id, paragraph in enumerate(text.split(FIL_SEP_PARA)):
+        output_sentences = []
+        for sentence in paragraph.split(FIL_SEP_SENT):
+            end = start + len(sentence)
+            # NEW: remove leading white spaces
+            lws = len(sentence) - len(sentence.lstrip())
+            if lws:
+                start += lws
+            # end NEW
+            if end > start:
+                output_sentences.append(Sentence(sent_id, Span(start, end)))
+                sent_id += 1
+            start = end + 3  # + 3 for + len(FIL_SEP_SENT)
+        output_paras.append(Paragraph(para_id, output_sentences))
+        start -= 1  # start += len(FIL_SEP_PARA) - len(FIL_SEP_SENT)
+    # TODO remove trailing '\n' of last sentence
+
+    return text, output_paras
+    
+
+def load_rst_wsj_corpus_text_file_file(f):
+    """Load a text file whose name is of the form `file##`
+
+    These files do not mark paragraphs.
+    Each line contains a sentence preceded by two or three leading spaces.
+    """
+    with codecs.open(f, 'r', 'utf-8') as f:
+        text, paragraphs = _load_rst_wsj_corpus_text_file_file(f)
+    return (text, paragraphs)
+
+
+##################
+# wsj_## files
+##################
+WSJ_SEP_PARA = ' \n\n'  # was: '\n\n'
+WSJ_SEP_SENT = '\n'
+
+
+def _load_rst_wsj_corpus_text_file_wsj(f):
+    """Actually do load"""
+    text = f.read()
+
+    start = 0
+    sent_id = 0
+    output_paras = []
+    for para_id, paragraph in enumerate(text.split(WSJ_SEP_PARA)):
+        output_sentences = []
+        for sentence in paragraph.split(WSJ_SEP_SENT):
+            end = start + len(sentence)
+            # NEW: remove trailing white space
+            rws = len(sentence) - len(sentence.rstrip())
+            if rws:
+                end -= rws
+            # end NEW
+            if end > start:
+                output_sentences.append(Sentence(sent_id, Span(start, end)))
+                sent_id += 1
+            start = end + rws + 1  # + 1 for + len(WSJ_SEP_SENT)
+        output_paras.append(Paragraph(para_id, output_sentences))
+        start += 2  # whitespace and second newline
+
+    return text, output_paras
+
```

Comment:
In addition to different separators, each type of file needs special extra cleaning to handle its own dirty irregularities. fileN uses 2 leading whitespaces but can contain more without any particular reason, wsj_ can have trailing whitespaces...
The cleaning process is different enough that I prefer to keep separate functions.
In the near future they should diverge even more with more cleaning (e.g. removing intra-sentential newlines in fileN).